javascripts/govuk/analytics/tracker.js

```diff
@@ -0,0 +1,81 @@
+(function() {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  var Tracker = function(config) {
+    var setDimension = this.setDimension.bind(this);
+
+    this.universal = new GOVUK.GoogleAnalyticsUniversalTracker(config.universalId, config.cookieDomain);
+    this.classic = new GOVUK.GoogleAnalyticsClassicTracker(config.classicId, config.cookieDomain);
+
+    setPixelDensityDimension();
+    setHTTPStatusCodeDimension();
+
+    if (config.prePageviewConfiguration && typeof config.prePageviewConfiguration === 'function') {
+      config.prePageviewConfiguration();
+    }
+
+    this.trackPageview();
+
+    function setPixelDensityDimension() {
+      var pixelRatioDimensionIndex = 11;
+
+      if (window.devicePixelRatio) {
+        setDimension(pixelRatioDimensionIndex, window.devicePixelRatio, 'Pixel Ratio', 2);
+      }
+    }
+
+    function setHTTPStatusCodeDimension() {
+      var statusCode = window.httpStatusCode || 200,
+          statusCodeDimensionIndex = 15;
+
+      setDimension(statusCodeDimensionIndex, statusCode, 'httpStatusCode');
+    }
+
+  };
+
+  Tracker.load = function() {
+    GOVUK.GoogleAnalyticsClassicTracker.load();
+    GOVUK.GoogleAnalyticsUniversalTracker.load();
+  };
+
+  Tracker.prototype.trackPageview = function(path, title) {
+    this.classic.trackPageview(path);
+    this.universal.trackPageview(path, title);
+  }
+
+  /*
+    https://developers.google.com/analytics/devguides/collection/analyticsjs/events
+    options.label – Useful for categorizing events (eg nav buttons)
+    options.value – Values must be non-negative. Useful to pass counts
+    options.nonInteraction – Prevent event from impacting bounce rate
+  */
+  Tracker.prototype.trackEvent = function(category, action, options) {
+    this.classic.trackEvent(category, action, options);
+    this.universal.trackEvent(category, action, options);
+  };
+
+  Tracker.prototype.trackShare = function(network) {
+    var target = location.pathname;
+    this.classic.trackSocial(network, 'share', target);
+    this.universal.trackSocial(network, 'share', target);
+  };
+
+  Tracker.prototype.setDimension = function(index, value, name, scope) {
+    var PAGE_LEVEL_SCOPE = 3;
+    scope = scope || PAGE_LEVEL_SCOPE;
+
+    if (typeof index !== "number") {
+      index = parseInt(index, 10);
+    }
+
+    if (typeof scope !== "number") {
+      scope = parseInt(scope, 10);
+    }
+
+    this.universal.setDimension(index, value);
+    this.classic.setCustomVariable(index, value, name, scope);
```

Comment:
Can we just add this to the documentation rather than removing this. If someone else wants to use this and has different requirements we can have another look at how this is implemented but for the moment this will work for all the people that want this code.